### PR TITLE
Create Closed Loop Shooter Commands

### LINKED ISSUE #70: Shooter Set RPMs Command
A command to set the RPMs of the shooter.

Reference from 2022.
https://github.com/frc2357/FRC-2022/blob/main/robotbase/src/main/java/com/team2357/frc2022/commands/shooter/ShooterSetRPMsCommand.java

### PR DESCRIPTION
**Pull request by Tyson453**
Closes #70 and #72

### CODE REVIEW

File: robotbase/src/main/java/frc/robot/commands/pickup/SourceIntakeFromShooter.java [hunk 1]
```diff
@@ -1,14 +1,18 @@
...
 public class SourceIntakeFromShooter extends ParallelDeadlineGroup {
 
   public SourceIntakeFromShooter() {
     super(new IntakeReverseFeed());
+    addCommands(
+        new ShooterSetRPMs(
+            Constants.SHOOTER.TOP_MOTOR_SOURCE_INTAKE_RPMS,
+            Constants.SHOOTER.BOTTOM_MOTOR_SOURCE_INTAKE_RPMS));
     addCommands(new SourceIntakeFromShooter());
-    addCommands(new ShooterPivotIntakeFromSource());
```
Comment (NolanCampbell2357) [inline robotbase/src/main/java/frc/robot/commands/pickup/SourceIntakeFromShooter.java:12]: Why did you remove this, are you sure we won't need to rotate to pick up from the source?

File: robotbase/src/main/java/frc/robot/subsystems/Shooter.java [hunk 2]
```diff
@@ -103,14 +114,11 @@ public void periodic() {
...
 
-  public void startVisionShooting() {
-    m_isClosedLoopEnabled = true;
-    Robot.shooterLimelight.setPipeline(Constants.SHOOTER_LIMELIGHT.SPEAKER_PIPELINE_INDEX);
-  }
-
-  public void endVisionShooting() {
-    m_isClosedLoopEnabled = false;
-    Robot.shooterLimelight.setHumanPipelineActive();
+  public void setVisionShootingEnabled(boolean enabled) {
```
Comment (NolanCampbell2357) [inline robotbase/src/main/java/frc/robot/subsystems/Shooter.java:117]: We are probably going to want to expose a method to get if the shooter is in vision shooting mode. I would also consider moving vision shooting mode to robot state since we have other subsystem specific state values out there (drive type, drive target lock).

**Review by NolanCampbell2357 — CHANGES_REQUESTED** [review]

File: robotbase/src/main/java/frc/robot/subsystems/Shooter.java (hunk 2 above)
Comment (Tyson453) [reply robotbase/src/main/java/frc/robot/subsystems/Shooter.java:117]: Agreed. I was going to sit down and plan out the entire robot state soon (hopefully this weekend) and then implement it all once we have all of the main subsystems (shooter, intake, drive) fully made and into the main branch. Any other changes to the RobotState flow will be small enough to do later like a separate state for automatically climbing.

**Review by NolanCampbell2357 — APPROVED** [review]
